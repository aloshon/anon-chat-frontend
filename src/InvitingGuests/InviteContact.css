.invite-contact-card{
    margin-bottom: 10px;
}

@media (min-width: 900px) {
    .invite-contact-card {
        display:inline-block;
        width: 47%;
        padding-bottom: 20px;
        vertical-align: top;
        margin-left: 10px;
    }
}