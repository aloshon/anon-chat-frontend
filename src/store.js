import  { composeWithDevTools } from "redux-devtools-extension";
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";
import autoMergeLevel2 from "redux-persist/lib/stateReconciler/autoMergeLevel2";

import thunk from "redux-thunk";
import rootReducer from "./Reducers/rootReducer";
import { createStore, applyMiddleware } from "redux";

// only keep the darkMode reducer persistent in local storage
const persistConfig = {
  key: "root",
  whitelist:['darkMode'],
  storage,
  stateReconciler: autoMergeLevel2
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = createStore(
  persistedReducer,
  composeWithDevTools(
    applyMiddleware(thunk),
  )
);

export const persistedStore = persistStore(store);