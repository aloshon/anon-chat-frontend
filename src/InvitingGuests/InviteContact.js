import { useState, useEffect } from "react";
import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";
import AnonChatApi from '../api';
import "./InviteContact.css";
import {useSelector, useDispatch} from "react-redux";
import {invitedGuest} from "../Actions/actionCreators";

const InviteContact = ({unique_id, user_id, group_chat_id, nickname, username, guestList}) => {
    const [invited, setInvited] = useState(false);
    const darkMode = useSelector(state => state.darkMode);
    const dispatch = useDispatch();
    useEffect(() => {
        if(guestList){
            guestList.forEach(g => {
                if(g.user_id === user_id){
                    setInvited(true)
                } 
            })
            // only allow 10 users on the list
            if(guestList.length >= 10){
                setInvited(true)
            }
        }
    }, [user_id, guestList]);

    const handleInvite = async (evt) => {
        evt.preventDefault();
        try{
            if(window.confirm("Invite guest? Cannot be uninvited!")){
                try{
                    const res = await AnonChatApi.inviteGuest({unique_id, user_id, group_chat_id, username});
                    res.unique_id = unique_id;
                    invitedGuest(dispatch, res);
                } catch(e){
                    alert(`Cannot invite guest at this time. Please try again later.  ${e}`)
                }
            } else{
                return
            }
        } catch(e){
            alert(`ERROR INVITING GUEST: ${e}`)
        }
    }
    
    if(guestList){
        return (
            <Card className="invite-contact-card"
                style={{backgroundColor: darkMode.card}}>
                <Card.Body>
                    <Card.Title>{nickname}</Card.Title>
                    <Card.Text>Username: {username}</Card.Text>
                    {invited ? <Button variant="success" disabled={true}>Invited</Button> 
                    : <Button variant="success" onClick={handleInvite}>Invite!</Button>}
                </Card.Body>
            </Card>
        )
    }
    return <h1>Loading...</h1>
}

export default InviteContact;