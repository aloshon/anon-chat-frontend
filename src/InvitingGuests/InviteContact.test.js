import { render, screen } from '@testing-library/react';
import InviteContact from './InviteContact';
import { Provider } from "react-redux";
import { store } from "../store"
import {BrowserRouter} from 'react-router-dom';


describe("InviteContact should render contact info and whether if user is invited", () => {
    const guestList = [
        {user_id: 1}
    ]

    const {asFragment} = render(
      <Provider store={store}>
        <BrowserRouter>
            <InviteContact
            unique_id="04ffa447-5237-476e-8v24-fb8n0nc801c7"
            user_id={1}
            group_chat_id={1}
            nickname="nickname"
            username="user1"
            guestList={guestList}
            />
        </BrowserRouter>
      </Provider>
    );
    test("renders contact info and invite button", () => {
      const username = screen.getByText("Username: user1");
      const nickname = screen.getByText("nickname");
      const invited = screen.getByText("Invited");
      expect(username).toBeInTheDocument();
      expect(nickname).toBeInTheDocument();
      expect(invited).toBeInTheDocument();
    });
  
    test("matches snapshot", () => {
      expect(asFragment()).toMatchSnapshot();
    });
    
});

test("renders different text on button if user is not on guestList", () => {
    const guestList = [
        {user_id: 1}
    ]

    render(
      <Provider store={store}>
        <BrowserRouter>
            <InviteContact
            unique_id="04ffa447-5237-476e-8v24-fb8n0nc801c7"
            user_id={2}
            group_chat_id={1}
            nickname="nickname"
            username="user1"
            guestList={guestList}
            />
        </BrowserRouter>
      </Provider>
    );

    const username = screen.getByText("Username: user1");
    const nickname = screen.getByText("nickname");
    const notInvited = screen.getByText("Invite!");
    expect(username).toBeInTheDocument();
    expect(nickname).toBeInTheDocument();
    expect(notInvited).toBeInTheDocument();
});

