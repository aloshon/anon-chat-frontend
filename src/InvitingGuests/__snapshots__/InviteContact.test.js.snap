// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InviteContact should render contact info and whether if user is invited matches snapshot 1`] = `<DocumentFragment />`;
